tests/test_edgeql_expressions.py

```diff
@@ -879,6 +880,168 @@ async def test_edgeql_expr_mod_01(self):
+        await self.assert_query_result(
+            f'''select 2000000 % 9223372036854770000;''',
+            [2000000],
+        )
+        await self.assert_query_result(
+            f'''select -2000000 % -9223372036854770000;''',
+            [-2000000],
+        )
+        await self.assert_query_result(
+            f'''select (<int32>2000000) % <int32>2147483000;''',
+            [2000000],
+        )
+        await self.assert_query_result(
+            f'''select (-<int32>2000000) % -<int32>2147483000;''',
+            [-2000000],
+        )
+        await self.assert_query_result(
+            f'''select (<int16>20000) % <int16>32000;''',
+            [20000],
+        )
+        await self.assert_query_result(
+            f'''select (-<int16>20000) % -<int16>32000;''',
+            [-20000],
+        )
+
+    async def test_edgeql_expr_mod_03(self):
+        # Test integer division.
+        #
+        # We've had a bug where 100000000000000001 // 2 produced
+        # 50000000000000001. Basically the floor division sometimes
+        # rounded up.
+        await self.assert_query_result(
+            f'''select 100000000000000001 // 2;''',
+            [50000000000000000],
+        )
+        await self.assert_query_result(
+            f'''select -100000000000000001 // 2;''',
+            [-50000000000000001],
+        )
+        await self.assert_query_result(
+            f'''select (<int32>1000000001) // <int32>2;''',
+            [500000000],
+        )
+        await self.assert_query_result(
+            f'''select (-<int32>1000000001) // <int32>2;''',
+            [-500000001],
+        )
+        await self.assert_query_result(
+            f'''select (<int16>10001) // <int16>2;''',
+            [5000],
+        )
+        await self.assert_query_result(
+            f'''select (-<int16>10001) // <int16>2;''',
+            [-5001],
+        )
+        await self.assert_query_result(
+            f'''select 10000000000000000000000000000000001n // 2n;''',
+            [5000000000000000000000000000000000],
+        )
+        await self.assert_query_result(
+            f'''select -10000000000000000000000000000000001n // 2n;''',
+            [-5000000000000000000000000000000001],
+        )
+        await self.assert_query_result(
+            f'''select 10000000000000000000000000000000001.0n // 2.0n;''',
+            [5000000000000000000000000000000000],
+        )
+        await self.assert_query_result(
+            f'''select -10000000000000000000000000000000001.0n // 2.0n;''',
+            [-5000000000000000000000000000000001],
+        )
+
+    async def test_edgeql_expr_mod_04(self):
+        # Fuzztest of int64 //
+        MAX = 9223372036854775807
+        vals = []
+        for i in range(500):
+            a = random.randrange(-MAX, MAX)
+            b = random.randrange(-MAX, MAX)
+            vals.append([i, a, b, a // b])
+
+        await self.assert_query_result(
```

Comment:
`assert_query_result` takes a `variables` arg; maybe it would be better to use that instead of building a big query?

Reply:
Yeah, I've used variables and it also sped things up drastically.